temporal-sdk/src/main/java/io/temporal/workflow/SystemSearchAttributes.java

```diff
@@ -17,21 +17,32 @@
  *  permissions and limitations under the License.
  */
 
-package io.temporal.internal.common;
+package io.temporal.workflow;
 
-import com.cronutils.utils.StringUtils;
-import io.temporal.api.common.v1.SearchAttributes;
-import io.temporal.common.converter.DataConverter;
-
-public class SearchAttributesUtil {
-  private static final DataConverter jsonConverter = DataConverter.getDefaultInstance();
-
-  public static <T> T getValueFromSearchAttributes(
-      SearchAttributes searchAttributes, String key, Class<T> classType) {
-    if (searchAttributes == null || StringUtils.isEmpty(key)) {
-      return null;
-    }
-    return jsonConverter.fromPayload(
-        searchAttributes.getIndexedFieldsOrThrow(key), classType, classType);
-  }
+/**
+ * These are the system attributes that are provided by default. Use {@link #name()} to get a key
+ * for our Search Attributes objects map.
+ */
+public enum SystemSearchAttributes {
+  BatcherNamespace,
```

Comment:
It looks like actually most of these values are never in the SearchAtttibutes object at all.
Only TemporalChangeVersion, BinaryChecksums, BatcherNamespace, BatcherUser are there.
Everything else is actually not here.
Everything that is defined in this enum on server side: https://github.com/temporalio/temporal/blob/36cd514da6a1333bcebd7c45bb99314a0e786bd0/common/searchattribute/defs.go#L57
is not part of SearchAttributes. I'm already not a big fan of having any logic related to these predefined attributes at all.
Let's maybe kill it altogether, especially if we decided that it's ok to trust "type" metadata?